source/extensions/dynamic_modules/sdk/rust/src/lib.rs

```diff
@@ -171,9 +270,110 @@ unsafe extern "C" fn envoy_dynamic_module_on_http_filter_config_destroy(
+}
+
+#[no_mangle]
+unsafe extern "C" fn envoy_dynamic_module_on_http_filter_request_headers(
+  envoy_ptr: abi::envoy_dynamic_module_type_http_filter_envoy_ptr,
+  filter_ptr: abi::envoy_dynamic_module_type_http_filter_module_ptr,
+  end_of_stream: bool,
+) -> abi::envoy_dynamic_module_type_on_http_filter_request_headers_status {
+  let filter = filter_ptr as *mut *mut dyn HttpFilter;
+  let filter = &mut **filter;
+  filter.on_request_headers(EnvoyHttpFilter { raw_ptr: envoy_ptr }, end_of_stream)
+}
+
+#[no_mangle]
+unsafe extern "C" fn envoy_dynamic_module_on_http_filter_request_body(
+  envoy_ptr: abi::envoy_dynamic_module_type_http_filter_envoy_ptr,
+  filter_ptr: abi::envoy_dynamic_module_type_http_filter_module_ptr,
+  end_of_stream: bool,
+) -> abi::envoy_dynamic_module_type_on_http_filter_request_body_status {
+  let filter = filter_ptr as *mut *mut dyn HttpFilter;
+  let filter = &mut **filter;
+  filter.on_request_body(EnvoyHttpFilter { raw_ptr: envoy_ptr }, end_of_stream)
+}
+
+#[no_mangle]
+unsafe extern "C" fn envoy_dynamic_module_on_http_filter_request_trailers(
+  envoy_ptr: abi::envoy_dynamic_module_type_http_filter_envoy_ptr,
+  filter_ptr: abi::envoy_dynamic_module_type_http_filter_module_ptr,
+) -> abi::envoy_dynamic_module_type_on_http_filter_request_trailers_status {
+  let filter = filter_ptr as *mut *mut dyn HttpFilter;
+  let filter = &mut **filter;
+  filter.on_request_trailers(EnvoyHttpFilter { raw_ptr: envoy_ptr })
+}
+
+#[no_mangle]
+unsafe extern "C" fn envoy_dynamic_module_on_http_filter_response_headers(
+  envoy_ptr: abi::envoy_dynamic_module_type_http_filter_envoy_ptr,
+  filter_ptr: abi::envoy_dynamic_module_type_http_filter_module_ptr,
+  end_of_stream: bool,
+) -> abi::envoy_dynamic_module_type_on_http_filter_response_headers_status {
+  let filter = filter_ptr as *mut *mut dyn HttpFilter;
+  let filter = &mut **filter;
+  filter.on_response_headers(EnvoyHttpFilter { raw_ptr: envoy_ptr }, end_of_stream)
+}
+
+#[no_mangle]
+unsafe extern "C" fn envoy_dynamic_module_on_http_filter_response_body(
+  envoy_ptr: abi::envoy_dynamic_module_type_http_filter_envoy_ptr,
+  filter_ptr: abi::envoy_dynamic_module_type_http_filter_module_ptr,
+  end_of_stream: bool,
+) -> abi::envoy_dynamic_module_type_on_http_filter_response_body_status {
+  let filter = filter_ptr as *mut *mut dyn HttpFilter;
+  let filter = &mut **filter;
+  filter.on_response_body(EnvoyHttpFilter { raw_ptr: envoy_ptr }, end_of_stream)
+}
+
+#[no_mangle]
+unsafe extern "C" fn envoy_dynamic_module_on_http_filter_response_trailers(
+  envoy_ptr: abi::envoy_dynamic_module_type_http_filter_envoy_ptr,
+  filter_ptr: abi::envoy_dynamic_module_type_http_filter_module_ptr,
+) -> abi::envoy_dynamic_module_type_on_http_filter_response_trailers_status {
+  let filter = filter_ptr as *mut *mut dyn HttpFilter;
+  let filter = &mut **filter;
+  filter.on_response_trailers(EnvoyHttpFilter { raw_ptr: envoy_ptr })
+}
+
 #[cfg(test)]
 mod tests {
```

Comment:
This is a personal Rust nit of mine which is not standard, but I prefer tests be kept in separate files as it makes it much easier to grep/search/exclude. Can you make a lib_test that sits next to this file and put the tests there. Please do this for other Rust code we add.

Reply:
first time to hear that pattern, and actually that sounds reasonable.  done in https://github.com/envoyproxy/envoy/pull/37325/commits/74cacdb5bd25259d70bc5073768264a12d88f351 following the pattern of your repository: https://github.com/bitdriftlabs/pulse/blob/46bebc3b55fc8694dc66cf823e0b9c938b960416/pulse-metrics/src/pipeline/processor/elision/mod.rs#L8-L10